src/channel/NodeChannel.js

```diff
@@ -18,73 +18,151 @@
  * ‍
  */
 
+import tls from "tls";
 import { Client, credentials } from "@grpc/grpc-js";
 import Channel from "./Channel.js";
 import GrpcServicesError from "../grpc/GrpcServiceError.js";
 import GrpcStatus from "../grpc/GrpcStatus.js";
 import { ALL_NETWORK_IPS } from "../constants/ClientConstants.js";
 
-/**
- * @property {?HashgraphProto.proto.CryptoService} _crypto
- * @property {?HashgraphProto.proto.SmartContractService} _smartContract
- * @property {?HashgraphProto.proto.FileService} _file
- * @property {?HashgraphProto.proto.FreezeService} _freeze
- * @property {?HashgraphProto.proto.ConsensusService} _consensus
- * @property {?HashgraphProto.proto.NetworkService} _network
- */
+/** @type {{ [key: string]: string }} */
+const certificateCache = {};
+/** @type {{ [key: string]: Client }} */
+const clientCache = {};
```

Comment:
Do we need the `clientCache`? If we have the cert in cache, the only step we skip with the addition of the `clientCache` is `security = credentials.createSsl(certificate);` OR `security = credentials.createInsecure();`

Reply:
so we skip the `credentials.createSsl(certificate)` OR `credentials.createInsecure()` which are not doing any call through the network and we are keeping X-number of `Client` objects in memory in return. 
 The client is quite a big object, in my opinion it's not worth to keep it in memory.

Reply:
This `client` is actually a channel client from `@grpc/grpc-js` rather than the SDK client. I will remove the certificateCache as its pointless in the situation